rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java

```diff
@@ -313,6 +317,28 @@ public static boolean isParallelWarmupEnabled(Configuration conf)
     return conf.getBoolean(KEY_PARALLEL_WARMUP, DEFAULT_PARALLEL_WARMUP);
   }
 
+  public static String getPrestoClusterManager(Configuration conf)
+  {
+    return conf.get(KEY_PRESTO_CLUSTER_MANAGER, DEFAULT_PRESTO_CLUSTER_MANAGER);
+  }
+
+  public static String getHadoopClusterManager(Configuration conf)
+  {
+    return conf.get(KEY_HADOOP_CLUSTER_MANAGER, DEFAULT_HADOOP_CLUSTER_MANAGER);
+  }
+
+  public static String getClusterManagerClass(Configuration conf, ClusterType clusterType)
+  {
+    switch (clusterType) {
```

Comment:
Can the callers use the right keys ? The enum seems extraneous to me.

Reply:
The enum already exists. The caller will decide whether it needs a presto cluster manager or hadoop2 cluster manager

Reply:
Hmm - the available ways for specialization are:
1. CachingFileSystem -> [PrestoCachingFileSystem, Hadoop2CachingFileSystem]
2. ClusterType
3. ClusterManager -> [PrestoClusterManager, Hadoop2ClusterManager]

What you need is a mapping between 1 & 3. You are using 2. which seems unnecessary.

I prefer the following which is a direct mapping between 1 & 3:

    class CachingFileSystem {
        abstract String getClusterManagerClass();
     }

     class PrestoCachingFileSystem {
        String getClusterManagerClass() { return CacheConfig.getPrestoClusterManager();}
     }

Actually there doesnt seem to be much use of 2. I checked for usages of `getClusterType` and I didnt find anything interesting.

Reply:
In BookKeeper server, we need to get cluster manager to get the nodes. And the cluster manager object is set only once for the server. BookKeeper doesn't have reference of CachingFileSystem (it should not have it). In that case we create the cluster manager object based on the cluster type enum. Thats why I took approach 2. We cannot have [getClusterManagerClass](url) as abstract method then in `CachingFileSystem`.

Reply:
OK. I get it. This goes back to `Bookkeeper.readData`. The clients tell who they are. I prefer starting the rubix server with a particular type but thats a separate debate.